premium/ecommerce/gdpr-send-to-mailchimp.php

```diff
@@ -5,38 +5,99 @@
  */
 
 /**
- * Send a request to mailchimp to check if member has status subscribed or pending
+ * Check if a list member has status pending or subscribed based on there email
  *
  * @see https://mailchimp.com/developer/reference/lists/list-members/
  *
  * @param string $email
  *
  * @return bool
  */
-function sync_only_subscribed_or_pending_emails($email) {
+function mailchimp_list_member_has_status_pending_or_subscribed($email) {
+
 	if(!is_string($email)) {
 		throw new InvalidArgumentException(
-			sprintf("sync_only_subscribed_or_pending_emails() expects parameter 1 to be array or %s, %s given", Traversable::class, gettype($emails))
+			sprintf("mailchimp_list_member_has_status_pending_or_subscribed() expects parameter 1 to be string, %s given", gettype($email))
 		);
 	}
 
-	if(!mc4wp()->has('ecommerce.options')) {
-		return false;
-	}
+    if(!mc4wp()->has('ecommerce.options')) {
+        return false;
+    }
 
-	$options = mc4wp('ecommerce.options');
+    $options = mc4wp('ecommerce.options');
 
-	if (!isset($options['store']['list_id'])) {
-		return false;
-	}
+    if (!isset($options['store']['list_id'])) {
+        return false;
+    }
+    //get the list ecommerce. Make it static becuase
+    $list = $options['store']['list_id'];
```

Comment:
Shall we name this `mailchimp_list_id` to make it super clear what this variable is?